src/yetibot/core/util/config.clj

```diff
@@ -41,26 +41,23 @@
   "Lookup configuration in a config tree.
    Returns one of:
    {:error :invalid
-    :message \"$value does not validate against schema: $schema\"
-    :schema $schema}
+    :message \"$value does not validate against spec: $spec\"
+    :spec $spec}
    {:error :not-found :message $path}
    {:value $valid-value}"
-  [config schema path]
-  ;; store up all schemas to build a complete schema-by-path representation as
+  [config spec path]
+  ;; store up all specs to build a complete spec-by-path representation as
   ;; a convenient reference
-  (swap! schema-by-path assoc path schema)
-  (trace "get-config" schema path)
+  (swap! spec-by-path assoc path spec)
+  (trace "get-config" spec path)
   (let [path (if (coll? path) path [path])]
     (if-let [value (get-in config path)]
-      (try
-        (s/validate schema value)
+      (if (s/valid? spec value)
         ;; valid and found
         {:value value}
-        (catch Exception e
-          ;; invalid against schema
-          ;; (warn e)
-          {:error :invalid
-           :message (str value " does not validate against schema: " (pr-str schema))
-           :schema schema}))
+        ;; invalid against spec
+        {:error :invalid
+         :message (str value " does not validate against spec: " (pr-str spec))
```

Comment:
We could use https://github.com/bhb/expound (or something like it) to provide a better error message here.

Reply:
Thanks for suggesting this, never heard of it before. Tried it and I think the output looks good.